src/accountsdb/snapshots.zig

```diff
@@ -164,50 +359,287 @@ pub const HashAge = struct {
+        const hard_forks_len = rand.uintAtMost(usize, max_list_entries);
+
+        const hard_forks = try allocator.alloc(SlotAndCount, hard_forks_len);
+        errdefer allocator.free(hard_forks);
+
+        for (hard_forks) |*hard_fork| hard_fork.* = .{
+            rand.int(Slot),
+            rand.int(usize),
+        };
+
+        return .{ .items = hard_forks };
+    }
 };
 
 /// Analogous to [NodeVoteAccounts](https://github.com/anza-xyz/agave/blob/574bae8fefc0ed256b55340b9d87b7689bcdf222/runtime/src/epoch_stakes.rs#L14)
 pub const NodeVoteAccounts = struct {
-    vote_accounts: ArrayList(Pubkey),
+    vote_accounts: std.ArrayListUnmanaged(Pubkey),
     total_stake: u64,
+
+    pub fn deinit(node_vote_accounts: NodeVoteAccounts, allocator: std.mem.Allocator) void {
+        var vote_accounts = node_vote_accounts.vote_accounts;
+        vote_accounts.deinit(allocator);
+    }
+
+    pub fn random(
+        rand: std.Random,
+        allocator: std.mem.Allocator,
+        max_list_entries: usize,
+    ) std.mem.Allocator.Error!NodeVoteAccounts {
+        const vote_accounts = try allocator.alloc(Pubkey, rand.uintLessThan(usize, max_list_entries));
+        errdefer allocator.free(vote_accounts);
+        for (vote_accounts) |*vote_account| vote_account.* = Pubkey.random(rand);
+        return .{
+            .vote_accounts = std.ArrayListUnmanaged(Pubkey).fromOwnedSlice(vote_accounts),
+            .total_stake = rand.int(u64),
+        };
+    }
 };
 
+pub const NodeIdToVoteAccountsMap = std.AutoArrayHashMapUnmanaged(Pubkey, NodeVoteAccounts);
+pub fn nodeIdToVoteAccountsMapDeinit(map: NodeIdToVoteAccountsMap, allocator: std.mem.Allocator) void {
+    for (map.values()) |*node_vote_accounts| {
+        node_vote_accounts.deinit(allocator);
+    }
+    var copy = map;
+    copy.deinit(allocator);
+}
+
+pub fn nodeIdToVoteAccountsMapRandom(
+    rand: std.Random,
+    allocator: std.mem.Allocator,
+    max_list_entries: usize,
+) std.mem.Allocator.Error!NodeIdToVoteAccountsMap {
+    var node_id_to_vote_accounts = NodeIdToVoteAccountsMap.Managed.init(allocator);
+    errdefer nodeIdToVoteAccountsMapDeinit(node_id_to_vote_accounts.unmanaged, allocator);
+
+    try sig.rand.fillHashmapWithRng(&node_id_to_vote_accounts, rand, rand.uintAtMost(usize, max_list_entries), struct {
+        allocator: std.mem.Allocator,
+        max_list_entries: usize,
+
+        pub fn randomKey(_: @This(), _rand: std.Random) !Pubkey {
+            return Pubkey.random(_rand);
+        }
+
+        pub fn randomValue(ctx: @This(), _rand: std.Random) !NodeVoteAccounts {
+            return try NodeVoteAccounts.random(_rand, ctx.allocator, ctx.max_list_entries);
+        }
+    }{
+        .allocator = allocator,
+        .max_list_entries = max_list_entries,
+    });
+
+    return node_id_to_vote_accounts.unmanaged;
+}
+
+pub const EpochAuthorizedVoters = std.AutoArrayHashMapUnmanaged(Pubkey, Pubkey);
+pub fn epochAuthorizedVotersRandom(
```

Comment:
are we sure we dont want a blank space between the type and the method - i get how it can tie the type + method together (but can lose its value when there are more than one method associated with the type - (random & deinit)) - maybe we can add a comment like 
```
/// available methods:
/// - epochAuthorizedVotersRandom
pub const EpochAuthorizedVoters = std.AutoArrayHashMapUnmanaged(Pubkey, Pubkey);

pub fn epochAuthorizedVotersRandom()
```

Reply:
Yeah that's fair, I don't think it needs a comment, being prefixed with the same name is a fine enough association I think, and probably less brittle than a comment.

Reply:
d9aaa44075d78ac24c879443138918fb34e53895